Automattic/sensei
# Add theme compatibility to the course results page

Partial solution to #2154

This PR takes the solution in #2205 and abstracts the functionality that imitates a single post/page so that it can be used in the course results view (and later in taxonomy archive views). 

## Still to Do
- [x] Adapt and create new tests

## Testing
- Go through testing steps from #2205 and make sure nothing is broken.
- Create a course with lessons.
- Sign up for the course and then visit course results page (adding `/results` to the course page; e.g. `http://sensei.docker/course/dog-whispering/results`).
- Ensure that the page looks right on themes that support Sensei, and themes that do not support Sensei.
- For themes that do not support Sensei, ensure that:
  - The title is not duplicated.
  - The pagination does NOT show up.
